### PR TITLE
chore: log filtered transactions and fix gas limits in tests

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

This PR adds loggers back to the filter functions (but actually uses them this time), and logs when transaction get filtered during `PrepareProposal`. It also re-adds an obtuse gas limit to the transactions in the `TestPrepareProposalConsistency` test, where those errors were getting caught by the filter functions. 

part of #2517

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: app/validate_txs.go [hunk 1]
```diff
@@ -55,7 +58,7 @@ func filterStdTxs(dec sdk.TxDecoder, ctx sdk.Context, handler sdk.AnteHandler, t
 // filterBlobTxs applies the provided antehandler to each transaction
 // and removes transactions that return an error. Panics are caught by the checkTxValidity
 // function used to apply the ante handler.
-func filterBlobTxs(dec sdk.TxDecoder, ctx sdk.Context, handler sdk.AnteHandler, txs []tmproto.BlobTx) ([]tmproto.BlobTx, sdk.Context) {
+func filterBlobTxs(logger log.Logger, dec sdk.TxDecoder, ctx sdk.Context, handler sdk.AnteHandler, txs []tmproto.BlobTx) ([]tmproto.BlobTx, sdk.Context) {
```
Comment (evan-forbes) [inline app/validate_txs.go:61]: we are now hitting the unwritten 5 argument for a small function rule, so we might want to refactor this

File: app/test/fuzz_abci_test.go [hunk 2]
```diff
@@ -102,6 +103,7 @@ func TestPrepareProposalConsistency(t *testing.T) {
 						true,
 						testutil.ChainID,
 						accounts[:tt.count],
+						user.SetGasLimitAndFee(1_000_000_000, 0.1),
```
Comment (evan-forbes) [inline app/test/fuzz_abci_test.go:106]: [for reviewers]
this fix could be separated out into its own PR. happy to do so if reviewers would appreciate that. originally went to go fix this issue and added the loggers simply to check if this change would fix the issue.

**Comment by evan-forbes** [comment]
also see https://github.com/celestiaorg/celestia-app/issues/2519

**Review by rootulp — APPROVED** [review]

File: app/validate_txs.go [hunk 3]
```diff
@@ -42,6 +44,7 @@ func filterStdTxs(dec sdk.TxDecoder, ctx sdk.Context, handler sdk.AnteHandler, t
 		// simply want to remove this tx, or we're catching a panic from one
 		// of the anteHanders which is logged.
 		if err != nil {
+			logger.Error("filtering already checked transaction", "tx", tmbytes.FromBytes(tx), "error", err)
```
Comment (cmwaters) [inline app/validate_txs.go:47]: Ideally would prefer not to dump the entire transaction. It wouldn't really help for debugability. I think it would be better either to decode it and print it out or just to print out the message type

Comment (evan-forbes) [reply app/validate_txs.go:47]: whoops good catch mean to hash that

Comment (evan-forbes) [reply app/validate_txs.go]: logging the hash and the msg types for the sdk txs

https://github.com/celestiaorg/celestia-app/pull/2518/commits/12eed6d464767cc1e773b97dc3c5b0639558733b

e486596441113fb01f318ba4d33a7edb85dceec1

**Review by rootulp — APPROVED** [review]

**Review by cmwaters — APPROVED** [review]
